internal/cli/approve.go

```diff
@@ -116,15 +117,22 @@ func (o *ApproveOptions) Run(ctx context.Context, args []string) error {
 	return processApprovalReponse(response, err, kind, name)
 }
 
-func processApprovalReponse(response interface{}, err error, kind string, name string) error {
+func processApprovalReponse(response *http.Response, err error, kind string, name string) error {
 	errorPrefix := fmt.Sprintf("approving %s/%s", kind, name)
 	if err != nil {
 		return fmt.Errorf("%s: %w", errorPrefix, err)
 	}
-
-	v := reflect.ValueOf(response).Elem()
-	if v.FieldByName("StatusCode").Int() != http.StatusOK {
-		return fmt.Errorf(errorPrefix+": %s (%d)", v.FieldByName("Status").String(), v.FieldByName("StatusCode").Int())
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		var responseError api.Error
+		// not handling errors as we are only interested in the message
+		// and in case there will be a problem in reading the body or unmarshalling it
+		// we will print the status like
+		// Error: approving enrollmentrequest/<name>:  (422 Unprocessable Entity)
+		body, _ := io.ReadAll(response.Body)
+		_ = json.Unmarshal(body, &responseError)
```

Comment:
why not handle this error?

Reply:
I actually thought that it is not relevant for user, I mean that it is better to return something more valuable than I am failing to unmarshal server response

Reply:
fair